src/persist-client/src/stats.rs

```diff
@@ -183,4 +190,15 @@ pub struct SnapshotStats {
     /// compaction never results in more updates than the sum of the inputs, it
     /// can only go down.
     pub num_updates: usize,
+    /// Stats for individual parts.
+    pub parts: Vec<SnapshotPartStats>,
```

Comment:
Not particularly a fan of reusing the `snapshot_stats` call for this feature. That call is meant to be super lightweight for the optimizer (which this add an alloc and such) and is allowed to be approximate (which is bad for the explain feature).

My initial reaction is that we should be using `ReadHandle::snapshot`, even if means we have to register and expire one as part of the explain execution

Reply:
I (believe I) have pushed more or less what you suggest here!